pkg/clusteragent/admission/controllers/webhook/controller_base.go

```diff
@@ -39,18 +45,62 @@ func NewController(client kubernetes.Interface, secretInformer coreinformers.Sec
 	return NewControllerV1beta1(client, secretInformer, admissionInterface.V1beta1().MutatingWebhookConfigurations(), isLeaderFunc, isLeaderNotif, config)
 }
 
+// MutatingWebhook represents a mutating webhook
+type MutatingWebhook interface {
+	// Name returns the name of the webhook
+	Name() string
+	// IsEnabled returns whether the webhook is enabled
+	IsEnabled() bool
+	// Endpoint returns the endpoint of the webhook
+	Endpoint() string
+	// Resources returns the kubernetes resources for which the webhook should
+	// be invoked
+	Resources() []string
+	// Operations returns the operations on the resources specified for which
+	// the webhook should be invoked
+	Operations() []admiv1.OperationType
+	// LabelSelectors returns the label selectors that specify when the webhook
+	// should be invoked
+	LabelSelectors(useNamespaceSelector bool) (namespaceSelector *metav1.LabelSelector, objectSelector *metav1.LabelSelector)
+	// MutateFunc returns the function that mutates the resources
+	MutateFunc() admission.Func
+}
+
+func mutatingWebhooks() []MutatingWebhook {
```

Comment:
I prefer to do this in a separate PR if you don't mind because there are quite a few calls to the config and not all of them are in the init functions.